protected/views/adminDataset/_form.php

```diff
@@ -643,6 +643,32 @@ class="form-control"
     </div><!-- /.modal-dialog -->
 </div><!-- /.modal -->
 <script>
+    $(document).ready(function () {
+        $('#mockup').on('click', function (event) {
+            event.preventDefault();
+
+            mockupUrl = $(this).attr("href");
+            form = $('#dataset-form');
+
+            $.ajax({
+                url: form.attr('action'),
+                method: 'POST',
+                data: form.serialize(),
+                success: function (response, textStatus, xhr) {
+                    if (200 === xhr.status) {
+                        window.location.href = mockupUrl;
+                    } else {
+                        $('#flashError').html('An error occurred while trying to save the dataset')
+                    }
+
+                },
+                error: function (error) {
+                    $('#flashError').html('An error occurred while trying to save the dataset')
+                }
+            })
+        });
+    });
+
```

Comment:
**praise:** This JS function/AJAX call makes sure to save any changes made in the form when clicking the `Create/Reset Private URL`, without refreshing or leaving the page. This can prevent loss of data and provide immediate feedback.
**praise:** Redirecting to the mockup page only after confirming the data in the form has been saved successfully.